servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ProcessorBufferUtils.java

```diff
@@ -17,22 +17,12 @@
 
 import io.servicetalk.concurrent.internal.TerminalNotification;
 
-import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
 import javax.annotation.Nullable;
 
 import static io.servicetalk.concurrent.api.SubscriberApiUtils.unwrapNullUnchecked;
-import static java.util.concurrent.atomic.AtomicReferenceFieldUpdater.newUpdater;
 
-abstract class AbstractProcessorBuffer {
-    private static final AtomicReferenceFieldUpdater<AbstractProcessorBuffer,
-            TerminalNotification> terminalUpdater = newUpdater(AbstractProcessorBuffer.class,
-            TerminalNotification.class, "terminal");
-
-    @Nullable
-    private volatile TerminalNotification terminal;
```

Comment:
@NiteshKant - are there any hard requirements for filtering out duplicate terminals before the queue, or does this change make sense?